dts/bindings/pinctrl/ambiq,apollo3-pinctrl.yaml

```diff
@@ -0,0 +1,122 @@
+  so you can modify it like this:
+
+  &pinctrl {
+    /* your modifications go here */
+  };
+
+  All device pin configurations should be placed in child nodes of the
+  'pinctrl' node, as shown in this example:
+
+  /* You can put this in places like a board-pinctrl.dtsi file in
+   * your board directory, or a devicetree overlay in your application.
+  */
+
+  /* include pre-defined combinations for the SoC variant used by the board */
+  #include <dt-bindings/pinctrl/ambiq-apollo3-pinctrl.h>
+
+  &pinctrl {
+    uart0_default: uart0_default {
+      group1 {
+        pinmux = <UART0TX_P60>;
+      };
+      group2 {
+        pinmux = <UART0RX_P47>;
+        input-enable;
+      };
+    };
+  };
+
+  The 'uart0_default' child node encodes the pin configurations for a
+  particular state of a device; in this case, the default (that is, active)
+  state.
+
+  As shown, pin configurations are organized in groups within each child node.
+  Each group can specify a list of pin function selections in the 'pinmux'
+  property.
+
+  A group can also specify shared pin properties common to all the specified
+  pins, such as the 'input-enable' property in group 2.
+
+compatible: "ambiq,apollo3-pinctrl"
+
+include: base.yaml
+
+child-binding:
+  description: |
+    Definitions for a pinctrl state.
+  child-binding:
+
+    include:
+      - name: pincfg-node.yaml
+        property-allowlist:
+          - input-enable
+          - drive-push-pull
+          - drive-open-drain
+          - bias-high-impedance
+          - bias-pull-up
+          - bias-pull-down
+
+    properties:
+      pinmux:
+        required: true
+        type: array
+        description: |
+          An array of pins sharing the same group properties. Each
+          element of the array is an integer constructed from the
+          pin number and the alternative function of the pin.
+      drive-strength:
+        type: string
+        enum:
+          - "0.1"
+          - "0.5"
+          - "0.75"
+          - "1.0"
+        default: "0.1"
+        description: |
+          The drive strength of a pin, relative to full-driver strength.
+          The default value is 0.1, which is the lowest one: 2mA.
+      ambiq,pull-up-ohms:
+        type: int
+        enum:
+          - 1500
+          - 6000
+          - 12000
+          - 24000
+        default: 1500
+        description: |
+          The 1.5K-24K pullup values are valid for select I2C enabled pads.
+          For Apollo3 these pins are 0-1,5-6,8-9,25,27,39-40,42-43,48-49.
+          The default value is 1500 ohms, which is the reset value of
+          register PADREGx.PADxRSEL.
+      ambiq,iom-nce-module:
+        type: int
+        default: 0
+        description: |
+          IOM nCE module select, selects the SPI channel (CE) number (0-3).
+          The default value is channel 0. If the pin is not a CE, this descriptor
+          will be ignored.
+      ambiq,iom-mspi:
+        type: int
+        default: 0
```

Comment:
also unsure but maybe this should be required

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/67815#discussion_r1578239958

Reply:
Justification added.